src/vabackend.c

```diff
@@ -1548,8 +1550,13 @@ VAStatus __vaDriverInit_1_0(VADriverContextP ctx)
     NVDriver *drv = (NVDriver*) calloc(1, sizeof(NVDriver));
     ctx->pDriverData = drv;
 
-    CHECK_CUDA_RESULT(cuInit(0));
-    CHECK_CUDA_RESULT(cuCtxCreate(&drv->cudaContext, CU_CTX_SCHED_BLOCKING_SYNC, 0));
+    cuda_load_functions(&cu, NULL);
+    cuvid_load_functions(&cv, NULL);
+    drv->cu = cu;
+    drv->cv = cv;
```

Comment:
Excuse my ignorance, but wouldn't it be better to load them directly into `drv`? What is the benefit of having those static variables? And shouldn't the return values be checked?

Reply:
Good point about checking the return values. I've updated the diff.

As for the static variables. There are some callsites, especially CHECK_CU_RESULT that need function access but where the driver isn't passed around. It could probable be wired through everywhere it's needed but that's pretty verbose and I didn't want to make that big of a change here.

The danger of statics here is that you can't safely initialise two instances of the driver in the same process, but that's not a relevant use-case for vaapi. Still, we could fix it if it was really necessary.

Reply:
> There are some callsites, especially CHECK_CU_RESULT that need function access but where the driver isn't passed around.

I see, I didn't notice that.

> The danger of statics here is that you can't safely initialise two instances of the driver in the same process

That was my main concern here.